client/src/partials/patients/record/patient_record.js

```diff
@@ -1,26 +1,61 @@
 angular.module('bhima.controllers')
 .controller('PatientRecordController', PatientRecordController);
 
-PatientRecordController.$inject = ['$stateParams', 'PatientService', 'NotifyService', 'moment'];
+PatientRecordController.$inject = ['$stateParams', 'PatientService', 'NotifyService', 'moment', 'Upload', '$timeout'];
 
-function PatientRecordController($stateParams, Patients, Notify, moment) {
+function PatientRecordController($stateParams, Patients, Notify, moment, Upload, $timeout) {
   var vm = this;
   var patientID = $stateParams.patientID;
   
   vm.loading = true;
+  vm.uploadFiles = uploadFiles;
+  vm.uploadButtonText = 'PATIENT_RECORDS.UPLOAD_PICTURE';
+ 
+  function uploadFiles(file, invalidFiles) {
+    if(invalidFiles.length){
+      Notify.danger('FORM.WARNNINGS.BAD_FILE_TYPE');  
+      return
+    }
+
+    if (file) {
+      var imageCheck = file.type.search("image/");
+      if(imageCheck !== -1){
+        file.upload = Upload.upload({
+          url: '/patients/' + patientID + '/pictures', 
+          data: {pictures: file}
+        });
+
+        file.upload.then(function (response) {
+          Notify.success('FORM.INFOS.PATIENT_SUCC_TRANSFERRED');
+          $timeout(function () {
+            vm.patient.avatar = response.data.link;
+          });
+        })
+        .catch(function (error) {
```

Comment:
In the future, this can the be `.catch(Notify.handleError);`.  There isn't a need to wrap this in a function.